frontend/src/pages/spotnet/position_history/PositionHistory.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import './position_history.css';
+import { ReactComponent as HealthIcon } from 'assets/icons/health.svg';
+import { ReactComponent as EthIcon } from 'assets/icons/ethereum.svg';
+import { ReactComponent as StrkIcon } from 'assets/icons/strk.svg';
+import { ReactComponent as UsdIcon } from 'assets/icons/usd_coin.svg';
+import { usePositionHistoryTable } from 'hooks/usePosition';
+import Spinner from 'components/spinner/Spinner';
+import { formatDate } from 'utils/formatDate';
+import useDashboardData from 'hooks/useDashboardData';
+import Card from 'components/Card/Card';
+import PositionHistoryModal from 'pages/spotnet/position_history/PositionHistoryModal';
+import { useWalletStore } from 'stores/useWalletStore';
+
+function PositionHistory() {
+  const { walletId } = useWalletStore();
+  const [healthFactor, setHealthFactor] = React.useState('0.00');
+  const [borrowed, setBorrowed] = React.useState('0.00');
+  const [selectedPosition, setSelectedPosition] = useState(null);
+  const { data: tableData, isLoading } = usePositionHistoryTable(walletId);
```

Comment:
'isLoading` doesn't exist in `useQuery` API - in order to check loading state it has `isPending`, please change and make sure loading works.